### ISSUE TITLE
Record with empty component list implied "not a record class"

### ISSUE THREAD

**Issue by commonquail** [avaje/avaje-record-builder#42]
JLS allows a record with an empty component list -- although such a record has... limited utility, it is legal.

Creating a record builder for a record with an empty component list is fairly useless and inane but not technically prohibited either.

If an avaje-record-build-1.0 `@RecordBuilder` is applied to a record with an empty component list the annotation processor generates an error and aborts, refusing to produce the requested builder. The record

```java
@io.avaje.recordbuilder.RecordBuilder
public record EmptyComponentList() {}
```

yields the error

```
error: Builders can only be generated for record classes
```

This error message is misleading. The JLS does not define a record as a class with a non-empty component list so the failure reason the annotation processor identifies is incorrect. [Class.isRecord()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Class.html#isRecord\(\)) informs whether a type is a record, and specifically raises the possibility that the component list may be empty.

I think refusing to produce the requested builder is an acceptable outcome per the above, if that's what avaje-record-builder wishes to do, but in that case with a dedicated error message. However, it is also technically a special casing, and even though a builder for an empty record appears at first useless I would not rule out use cases e.g. for consistent introspection. In any case, while I am convinced the current implementation is an accidental limitation (not a design decision), personally I would not bother to pursue such a design decision.

**Comment by SentryMan** [comment]
What a catch, that's a case I hadn't even considered. I think we'll just change the error message to say that the record must have components. I'm without a PC for the week so this is the only one I can fix at the moment.

**Comment by SentryMan** [comment]
Upon a second look, it seems you actually want to support empty records? I can add that.

**Comment by commonquail** [comment]
I was trying to say that it's not especially important to me, personally, how you choose to address this, but that if _I_ were the one to address it I reckon it would be by supporting empty records -- because, silly as that might be, that avoids inventing a special case on top of the JLS.

**Comment by rbygrave** [comment]
We can close this as fixed by #45
